torch/utils/_pytree.py

```diff
@@ -470,75 +402,102 @@ def _broadcast_to_and_flatten(pytree: PyTree, spec: TreeSpec) -> Optional[List[A
     return result
 
 
+"""
+TreeSpecSchema is the schema used to serialize the TreeSpec
+It contains the following fields:
+- type: A string name of the type. null for the case of a LeafSpec.
+- context: Any format which is json dumpable
+- children_spec: A list of children serialized specs.
+"""
+@dataclasses.dataclass
+class TreeSpecSchema:
+    type: Optional[str]
+    context: DumpableContext
+    children_spec: List['TreeSpecSchema']
+
```

Comment:
We should be careful what is public and what is not. For example, the user doesn't really need to know about TreeSpecSchema, it is an implementation detail, right?

At this point we can think of every non-underscore name in _pytree as public, with the exception of commonly used things like `_register_pytree_node`